frontend/src/utils/stringUtils.js

```diff
@@ -79,6 +79,29 @@ function displayCategory(arg) {
   }
 }
 
+/**
+ * Readable version of 'Set of {category}'
+ * @param {string} category node category
+ * @returns Set of {category}
+ */
+function setify(cat) {
+  let category = displayCategory(cat);
+  if (category.endsWith('ay')) {
```

Comment:
I did some of this heuristic declension manipulation in [mouse-trapi](https://github.com/ranking-agent/mouse-trapi). It's surprisingly painful.

Reply:
My goal here was to put in just enough logic so as to not be embarrassing, but to be able to move on to more important features.